netkan/netkan/metadata.py

```diff
@@ -121,3 +122,164 @@ def licenses(self):
+                        _result = ((str1 > str2)-(str1 < str2))
+                else:
+                    _result = ((str1 > str2)-(str1 < str2))
+
+                return _result, _first_remainder, _second_remainder
+
+            def _number_compare(v1, v2) -> (int, str, str):
+                _result: int
+                _first_remainder = ''
+                _second_remainder = ''
+
+                minimum_length1 = 0
+                for i in range(0, len(v1)):
+                    if not v1[i].isdigit():
+                        _first_remainder = v1[i:]
+                        break
+                    minimum_length1 += 1
+
+                minimum_length2 = 0
+                for i in range(0, len(v2)):
+                    if not v2[i].isdigit():
+                        _second_remainder = v2[i:]
+                        break
+                    minimum_length2 += 1
+
+                if v1[:minimum_length1].isnumeric():
+                    integer1 = int(v1[:minimum_length1])
+                else:
+                    integer1 = 0
+
+                if v2[:minimum_length2].isnumeric():
+                    integer2 = int(v2[:minimum_length2])
+                else:
+                    integer2 = 0
+
+                _result = ((integer1 > integer2)-(integer1 < integer2))
+                return _result, _first_remainder, _second_remainder
+
+            # Here begins the main comparison logic
+            if self is other:
+                return False
+
+            if other.epoch == self.epoch and other.bare_version == self.bare_version:
+                return False
+
+            # Compare epochs first
+            if self.epoch != other.epoch:
+                return self.epoch >= other.epoch
+
+            # Epochs are the same. Do the dance described in
+            # https://github.com/KSP-CKAN/CKAN/blob/master/Spec.md#version-ordering
+            first_remainder = self.bare_version
+            second_remainder = other.bare_version
+
+            # Process our strings while there are characters remaining
+            while len(first_remainder) > 0 and len(second_remainder) > 0:
+                # Start by comparing the string parts.
+                (result, first_remainder, second_remainder) = _string_compare(first_remainder, second_remainder)
+
+                if result != 0:
+                    return result > 0
+
+                # Otherwise, compare the number parts.
+                # It's okay not to check if our strings are exhausted, because
+                # if they are the exhausted parts will return zero.
+                (result, first_remainder, second_remainder) = _number_compare(first_remainder, second_remainder)
+
+                # Again, return difference if found.
+                if result != 0:
+                    return result > 0
+
+            # Oh, we've run out of one or both strings.
+            if len(first_remainder) == 0:
+                if len(second_remainder) == 0:
```

Comment:
This test looks redundant. If it's true, we'll return `False` in the next line. Otherwise we'll return `False on line 281. Either way we always return false.